### PR TITLE
Add set/get server version api

### PR DESCRIPTION
**Pull request by sunshine-Chun**
Summary: This diff added support for set/get server version api in rocksdb

Test Plan: Test it locally

Reviewers:

Subscribers:

Tasks:

Tags:

### CODE REVIEW

File: storage/rocksdb/rdb_datadic.h [hunk 1]
```diff
@@ -1725,6 +1730,10 @@ class Rdb_dict_manager : public Ensure_initialized {
   bool update_max_index_id(rocksdb::WriteBatch *const batch,
                            const uint32_t index_id,
                            bool is_dd_tbl = false) const;
+
+  bool get_server_version(uint *const version);
```
Comment (luqun) [inline storage/rocksdb/rdb_datadic.h]: nit: also const

File: storage/rocksdb/rdb_datadic.cc [hunk 2]
```diff
@@ -6033,6 +6038,40 @@ bool Rdb_dict_manager::update_max_index_id(rocksdb::WriteBatch *const batch,
...
+  }
+
+  return true;
+}
+
+bool Rdb_dict_manager::set_server_version() const {
+  const std::unique_ptr<rocksdb::WriteBatch> wb = begin();
+  rocksdb::WriteBatch *const batch = wb.get();
+
+  Rdb_buf_writer<Rdb_key_def::VERSION_SIZE + Rdb_key_def::INDEX_NUMBER_SIZE>
```
Comment (luqun) [inline storage/rocksdb/rdb_datadic.cc]: please don't use Rdb_key_def::INDEX_NUMBER_SIZE for value writer, although the value is same. you can define a new variable or ..

File: storage/rocksdb/rdb_datadic.cc [hunk 3]
```diff
@@ -6033,6 +6038,40 @@ bool Rdb_dict_manager::update_max_index_id(rocksdb::WriteBatch *const batch,
   return false;
 }
 
+bool Rdb_dict_manager::get_server_version(uint *const ser_version) const {
+  std::string value;
+
+  const rocksdb::Status status = get_value(m_key_slice_server_version, &value);
+  if (status.ok()) {
+    const uchar *const val = (const uchar *)value.c_str();
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.cc]: C++ style cast

File: storage/rocksdb/rdb_datadic.cc [hunk 4]
```diff
@@ -6033,6 +6038,40 @@ bool Rdb_dict_manager::update_max_index_id(rocksdb::WriteBatch *const batch,
...
+      return false;
+    }
+  }
+
+  return true;
+}
+
+bool Rdb_dict_manager::set_server_version() const {
+  const std::unique_ptr<rocksdb::WriteBatch> wb = begin();
+  rocksdb::WriteBatch *const batch = wb.get();
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_datadic.cc]: Why do we mirror a smart pointer in a raw pointer? IMHO removing the raw one, renaming `wb` to `batch`, and using `batch.get()` inline as needed would result in as clean or cleaner code

File: storage/rocksdb/ha_rocksdb.cc [hunk 5]
```diff
@@ -3257,6 +3257,25 @@ std::string dump_ingest_external_file_options(
   return s.str();
 }
 
+/**
```
Comment (laurynas-biveinis) [inline storage/rocksdb/ha_rocksdb.cc]: Please move to that new source file in the other PR

**Comment by sunshine-Chun** [comment]
dict_manager is a static variable declared in ha_rocksdb.cc. The get/set server version call rely on this variable. However this variable is not accessible in rdb_native_dd. we can either make it global or we can wrap it in ha_rocksdb. (The current implementation). LMK your thoughts.

File: storage/rocksdb/rdb_native_dd.h [hunk 6]
```diff
@@ -16,6 +16,7 @@
 #pragma once
 
 /* C++ standard header files */
+#include <sys/types.h>
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_native_dd.h:19]: Is this for `uint`? If so, forward-declare it instead

**Comment by laurynas-biveinis** [comment]
> dict_manager is a static variable declared in ha_rocksdb.cc. The get/set server version call rely on this variable. However this variable is not accessible in rdb_native_dd. we can either make it global or we can wrap it in ha_rocksdb. (The current implementation). LMK your thoughts.

Wrapping LGTM

File: storage/rocksdb/rdb_native_dd.h [hunk 7]
```diff
@@ -21,13 +21,17 @@
 /* MySQL header files */
 #include "sql/dd/object_id.h"
 
+typedef unsigned int uint;
```
Comment (laurynas-biveinis) [inline storage/rocksdb/rdb_native_dd.h]: I am sorry, I didn't realize you cannot forward declare a simple `typedef`. While repeating a `typedef` is correct, it's probably better to include the header in this case.

**Pull request closed without merging**